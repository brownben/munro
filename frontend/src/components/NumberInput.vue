<template>
  <div class="text-input top--4 relative appearance-none">
    <label
      class="font-heading text-sm text-main p-1 mx-3 bg-white relative select-none bottom--3 pb-0"
      >{{ label }}</label
    >
    <input
      :value="value"
      :min="min"
      :max="max"
      type="number"
      class="w-full border border-main rounded-tl-lg rounded-br-lg px-4 py-2 text-body outline-none appearance-none"
      @input="$emit('input', $event.target.value)"
    />
  </div>
</template>

<script>
export default {
  name: 'NumberInput',

  props: {
    value: {
      type: Number,
      default: 0,
    },
    label: {
      type: String,
      default: '',
    },
    min: {
      type: Number,
      default: 0,
    },
    max: {
      type: Number,
      default: 100,
    },
  },
}
</script>
<style>
.top--4 {
  top: -1rem;
}

.bottom--3 {
  bottom: -0.75rem;
}
</style>
