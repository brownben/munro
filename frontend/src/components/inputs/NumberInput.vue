<template>
  <div>
    <label
      class="block pb-1 text-gray-600 select-none font-heading"
      :for="label"
      >{{ label }}</label
    >
    <input
      :id="label"
      :value="value"
      :min="min"
      :max="max"
      type="number"
      class="w-full px-3 py-2 font-sans text-gray-900 transition duration-300 ease-in-out bg-white border outline-none appearance-none rounded-shape focus:shadow-outline focus:border-main"
      @input="$emit('input', $event.target.value)"
    />
  </div>
</template>

<script>
export default {
  name: 'NumberInput',

  props: {
    value: {
      type: Number,
      default: 0,
    },
    label: {
      type: String,
      default: '',
    },
    min: {
      type: Number,
      default: 0,
    },
    max: {
      type: Number,
      default: 100,
    },
  },
}
</script>
