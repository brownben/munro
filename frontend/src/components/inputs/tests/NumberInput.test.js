import { shallowMount } from '@vue/test-utils'
import NumberInput from '../NumberInput'

test('Renders Correctly', () => {
  const wrapper = shallowMount(NumberInput, {
    props: {
      label: 'Test Label:',
    },
  })
  expect(wrapper.exists()).toBeTruthy()
  expect(wrapper.find('label').text()).toBe('Test Label:')
  expect(wrapper.html()).toContain('<input')
  expect(wrapper.html()).toContain('type="number"')
  expect(wrapper.html()).toContain('min="0"')
  expect(wrapper.html()).toContain('max="100"')
})

test('Focus Is Correctly Applied to Label', async () => {
  const wrapper = shallowMount(NumberInput)
  expect(wrapper.find('label').classes('text-main-600')).toBeFalsy()
  expect(wrapper.find('label').classes('text-gray-600')).toBeTruthy()
  await wrapper.find('input').trigger('focus')
  expect(wrapper.find('label').classes('text-main-600')).toBeTruthy()
  expect(wrapper.find('label').classes('text-gray-600')).toBeFalsy()
  await wrapper.find('input').trigger('blur')
  expect(wrapper.find('label').classes('text-main-600')).toBeFalsy()
  expect(wrapper.find('label').classes('text-gray-600')).toBeTruthy()
})

test('Emits Data On Change', async () => {
  const wrapper = shallowMount(NumberInput)
  await wrapper.find('input').setValue(5)
  expect(wrapper.emitted()).toHaveProperty('update:modelValue')
  expect(wrapper.emitted()['update:modelValue'][0]).toEqual([5])
})
