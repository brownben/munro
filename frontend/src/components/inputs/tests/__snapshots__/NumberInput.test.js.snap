// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div>
  <label
    class="block pb-1 text-gray-600 select-none font-heading"
    for="Test"
  >
    Test
  </label>
   
  <input
    class="w-full px-3 py-2 font-sans text-gray-900 transition duration-300 ease-in-out bg-white border outline-none appearance-none rounded-shape focus:shadow-outline focus:border-main"
    id="Test"
    max="5"
    min="10"
    type="number"
  />
</div>
`;
